entgql/internal/todoplugin/cmd/entgqlgen.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019-present Facebook
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"entgo.io/contrib/entgql"
+	"entgo.io/contrib/entgql/plugin"
+	"entgo.io/ent/entc"
+	"entgo.io/ent/entc/gen"
+	"github.com/99designs/gqlgen/api"
+	"github.com/99designs/gqlgen/codegen/config"
+)
+
+func main() {
+	var (
+		schemaPath = flag.String("path", "", "path to schema directory")
+	)
+	flag.Parse()
+	if *schemaPath == "" {
+		log.Fatal("entgqlgen: must specify schema path. use entgqlgen -path ./ent/schema")
+	}
+
+	graph, err := entc.LoadGraph(*schemaPath, &gen.Config{})
+	if err != nil {
+		log.Fatalf("entgqlgen: failed loading ent graph: %v", err)
+	}
+
+	cfg, err := config.LoadConfigFromDefaultLocations()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "failed to load config", err.Error())
+		os.Exit(2)
+	}
+
+	entgqlPlugin, err := plugin.NewEntGQLPlugin(graph,
+		plugin.WithRelaySpecification(true),
+	)
+	if err != nil {
+		log.Fatalf("creating entgql plugin: %v", err)
+	}
+
+	ex, err := entgql.NewExtension()
+	if err != nil {
+		log.Fatalf("creating entgql extension: %v", err)
+	}
+	err = entc.Generate(*schemaPath, &gen.Config{
+		Header: `
+			// Copyright 2019-present Facebook
+			//
+			// Licensed under the Apache License, Version 2.0 (the "License");
+			// you may not use this file except in compliance with the License.
+			// You may obtain a copy of the License at
+			//
+			//      http://www.apache.org/licenses/LICENSE-2.0
+			//
+			// Unless required by applicable law or agreed to in writing, software
+			// distributed under the License is distributed on an "AS IS" BASIS,
+			// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+			// See the License for the specific language governing permissions and
+			// limitations under the License.
+			//
+			// Code generated by entc, DO NOT EDIT.
+		`,
+	}, entc.Extensions(ex))
+	if err != nil {
+		log.Fatalf("running ent codegen: %v", err)
+	}
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "failed to create entgql plugin", err.Error())
+		os.Exit(2)
+	}
+	err = api.Generate(cfg,
+		api.PrependPlugin(entgqlPlugin),
+	)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err.Error())
+		os.Exit(3)
```

Comment:
Why exit code 3?

Reply:
Not sure, it's come from the original PR. I can fix it by update to Fatal().